activerecord/lib/active_record/attributes.rb

```diff
@@ -247,7 +247,7 @@ def load_schema! # :nodoc:
         super
         attributes_to_define_after_schema_loads.each do |name, (type, options)|
           if type.is_a?(Symbol)
-            type = ActiveRecord::Type.lookup(type, **options.except(:default))
+            type = ActiveRecord::Type.lookup(type, adapter: connection.adapter_name, **options.except(:default))
```

Comment:
I was hesitant to simply call `connection` since I thought it might not be defined, instead of guarding it however, I figured since we are in the `load_schema` method and the `super` of that method has access to the `connection` I figured this was a safe assumption though it feels a little brittle.